### PR TITLE
[Docs/AIR] Various fixes for AIR examples

### PR DESCRIPTION
**Pull request by peytondmurray**
- Switch from deprecated CartPole-v0 to v1, fixing deprecation warnings in `rl_offline_example`, `rl_online_example`, and `rl_serving_example`.
- Fixed `feast_example` issues:
  - Numpy float warning
  - Deprecated `ray.ml` imports
  - Switch to downloading data to the current directory

## Why are these changes needed?

Addresses multiple deprecation warnings as well as a few issues which break the Feast example. See #27851 for more information.

## Related issue number

Partially addresses #27851.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-air/examples/feast_example.ipynb [hunk 1]
```diff
@@ -1132,103 +1187,61 @@
...
-      "  timesteps_since_restore: 0\n",
-      "  train-error: 0.01802706241815801\n",
-      "  train-logloss: 0.07058723671627426\n",
-      "  training_iteration: 81\n",
-      "  trial_id: a3a2c_00000\n",
-      "  validation-error: 0.0824022346368715\n",
-      "  validation-logloss: 0.22556984905196217\n",
-      "  warmup_time: 0.004194974899291992\n",
-      "  \n"
+      "/home/ray/Desktop/workspace/ray/python/ray/util/placement_group.py:78: DeprecationWarning: placement_group parameter is deprecated. Use scheduling_strategy=PlacementGroupSchedulingStrategy(...) instead, see the usage at https://docs.ray.io/en/master/ray-core/package-ref.html#ray-remote.\n",
```
Comment (richardliaw) [inline doc/source/ray-air/examples/feast_example.ipynb:1190]: hmm can we hide these?

Comment (richardliaw) [reply doc/source/ray-air/examples/feast_example.ipynb:1190]: wondering why these are showing up...

Comment (ericl) [reply doc/source/ray-air/examples/feast_example.ipynb:1190]: @jjyao seems we haven't finished this deprecation replacement.

Comment (peytondmurray) [reply doc/source/ray-air/examples/feast_example.ipynb:1190]: @richardliaw @ericl @jjyao These are fixed by #28315.

**Review by krfricke — APPROVED** [review]
Thanks for cleaning this up. Would you mind running it again on latest master since #28315 is merged to get rid of the deprecation warnings in the feast example?

Thanks!

**Comment by peytondmurray** [comment]
Okay, I think this should be good to go!

**Review by richardliaw — APPROVED** [review]

**Pull request merged**